### PR TITLE
DeviceInputSystem: Convert web implementation to use set/releasePointerCapture

### PR DESCRIPTION
**Pull request by PolygonalSun**
This PR changes the previous implementation of using the window as the event listening element back to the scene's provided HTMLElement.  Furthermore, setPointerCapture and releasePointerCapture are now being used to prevent lost of pointer button tracking.

These changes should also fix the following issue: https://github.com/BabylonJS/Babylon.js/issues/10353

### CODE REVIEW

File: src/DeviceInput/Implementations/webDeviceInputSystem.ts
Comment (deltakosh) [inline src/DeviceInput/Implementations/webDeviceInputSystem.ts:433]: Was it the case before? if no what is the rationale behind this choice?

Comment (darraghjburke) [reply src/DeviceInput/Implementations/webDeviceInputSystem.ts:433]: I believe this is because we want to capture pointerup events even if the pointer leaves the canvas before the user releases the button. Since the element captures the pointer, it will receive the pointerup event no matter where the pointer is on screen. https://developer.mozilla.org/en-US/docs/Web/API/Element/setPointerCapture

This is relevant to #10353  because the user may move their mouse outside the texture inspector window before releasing the pan button. The texture inspector need to catch this pointerup event and stop panning.

@PolygonalSun do I understand correctly?

File: src/DeviceInput/Implementations/webDeviceInputSystem.ts
Comment (darraghjburke) [inline src/DeviceInput/Implementations/webDeviceInputSystem.ts:44]: I think we may want to track a list of buttons we have captured, rather than assuming that only one pointer is pressed. If the user presses multiple mouse buttons down, we should capture them all, and then release them all when the related pointerup event occurs.

Comment (bghgary) [reply src/DeviceInput/Implementations/webDeviceInputSystem.ts:44]: Capturing the pointer should capture all pointer events to the element. I don't think we need to worry about which buttons. They will all be captured. If there are multiple mouse devices connected, then we may have an issue.

File: src/DeviceInput/Implementations/webDeviceInputSystem.ts
Comment (PolygonalSun) [reply src/DeviceInput/Implementations/webDeviceInputSystem.ts:433]: Yup, that's correct.  Initially, we were just listening to the window for pointerup events and handling it that way but that method doesn't work with situations that have multiple windows like the Texture Inspector.  This approach gives us identical behavior with single windows but also fixes pointerup listening for the pop out window.

Comment (deltakosh) [reply src/DeviceInput/Implementations/webDeviceInputSystem.ts:433]: Ok make sure it works when inside NME preview window, Playground, etc...

Comment (bghgary) [reply src/DeviceInput/Implementations/webDeviceInputSystem.ts:433]: There is an edge case where this doesn't work correctly. If you hold down two different buttons and let go of only one of them, the pointer capture will be released. Seems like we need to ref count for this to be totally correct.

File: src/DeviceInput/Implementations/webDeviceInputSystem.ts
Comment (PolygonalSun) [reply src/DeviceInput/Implementations/webDeviceInputSystem.ts:44]: For pointer button events, a "pointerup" event is only thrown all buttons are up.  Otherwise, a "pointermove" event is thrown when a button is released.

Things do get a little tricky when the cursor leaves the canvas with multiple buttons pressed though.  There is a standing bug (since 4.0) where dragging will "stick" if the cursor leaves the canvas with multiple buttons pressed.  It may be worth fleshing out the logic to try and fix this, though I might relegate that to a different PR so that Texture Inspector gets fixed first.

File: src/DeviceInput/Implementations/webDeviceInputSystem.ts
Comment (PolygonalSun) [reply src/DeviceInput/Implementations/webDeviceInputSystem.ts:433]: I wasn't able to repro this edge case result.  When I tried tracking this what I got was
FIRST BUTTON DOWN (pointerdown fired, capture set)
SECOND BUTTON DOWN (pointermove fired, capture still set)
ANY BUTTON UP (pointermove fired, capture still set)
OTHER BUTTON UP (pointerup fired, capture released)

The pointercapturelost event only fired after released in the pointerup (after the second button was released).  I've tried this with both the PG and Texture Inspector.  Is there a step that I might be missing to repro this?

Comment (bghgary) [reply src/DeviceInput/Implementations/webDeviceInputSystem.ts:433]: Sorry, I didn't actually try this code change. I mentioned this because I tried the [MDN sample](https://developer.mozilla.org/en-US/docs/Web/API/Element/setPointerCapture) and it didn't work right for this. Maybe this code works and I don't understand it well enough to see it.

**Review by darraghjburke — APPROVED** [review]
Looks good to me! Thanks for addressing the comments.

File: src/DeviceInput/Implementations/webDeviceInputSystem.ts
Comment (PolygonalSun) [reply src/DeviceInput/Implementations/webDeviceInputSystem.ts:433]: Just as an update to what this changes:
- NME: No change
- GUI Editor: No change
- Sandbox: PointerUp working in Texture Inspector window; Objects can be dragged off screen (drag coords can be read off window)
- Playground: Objects can be dragged off screen (drag coords can be read off window); works with Gizmos
While this is a slight change to behavior, I don't see this as a negative change.  The original behavior was that if the cursor left the screen while a button was pressed, the dragging behavior would stop but the object being dragged would "snap" to the coordinates of the cursor when it returned, assuming the button was never released.

Comment (deltakosh) [reply src/DeviceInput/Implementations/webDeviceInputSystem.ts:433]: This is pretty neat actually :D

**Review by deltakosh — APPROVED** [review]
Good job buddy! waiting for final approval by @bghgary and you can merge

**Review by bghgary — APPROVED** [review]
Good for me

**Pull request merged**
